test/Sentry.Unity.Editor.iOS.Tests/NativeMainTests.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.IO;
+using System.Reflection;
+using NUnit.Framework;
+
+namespace Sentry.Unity.Editor.iOS.Tests
+{
+    public class NativeMainTests
+    {
+        [Test]
+        public void ContainsSentry_SentryAlreadyAdded_ReturnsTrue()
+        {
+            var main = GetFileContents("main_expected.txt");
+            var nativeMain = new NativeMain();
+
+            var containsSentry = nativeMain.ContainsSentry(main, null);
+
+            Assert.IsTrue(containsSentry);
+        }
+
+        [Test]
+        public void ContainsSentry_SentryNotAdded_ReturnsFalse()
+        {
+            var main = GetFileContents("main.txt");
+            var nativeMain = new NativeMain();
+
+            var containsSentry = nativeMain.ContainsSentry(main, null);
+
+            Assert.IsFalse(containsSentry);
+        }
+
+        [Test]
+        public void AddSentryToMain_SentryNotAddedTo_MatchesExpectedOutput()
+        {
+            var main = GetFileContents("main.txt");
+            var expectedMain = GetFileContents("main_expected.txt");
+            var nativeMain = new NativeMain();
+
+            var actualMain = nativeMain.AddSentryToMain(main);
+
+            Assert.AreEqual(expectedMain, actualMain);
+        }
+
+        [Test]
+        public void AddSentryToMain_InvalidMain_ThrowsException()
+        {
+            var main = string.Empty;
+            var nativeMain = new NativeMain();
+
+            var x = Assert.Throws<ArgumentException>(() => nativeMain.AddSentryToMain(main));
+            Assert.AreEqual("main", x.ParamName);
+        }
+
+        [Test]
+        public void AddSentry_MainDoesNotExist_ThrowsFileNotFoundException()
```

Comment:
Test name indicates an exception is thrown but test is a copy paste from test below